src/server/game/Entities/Player/Player.cpp

```diff
@@ -11935,7 +11935,8 @@ void Player::VisualizeItem(uint8 slot, Item* pItem)
     if (pItem->GetTemplate()->GetBonding() == BIND_WHEN_EQUIPED || pItem->GetTemplate()->GetBonding() == BIND_WHEN_PICKED_UP || pItem->GetTemplate()->GetBonding() == BIND_QUEST_ITEM)
     {
         pItem->SetBinding(true);
-        GetSession()->GetCollectionMgr()->AddItemAppearance(pItem);
+        if (IsInWorld())
+            GetSession()->GetCollectionMgr()->AddItemAppearance(pItem);
```

Comment:
this change will not break the addition of the visualization. the appearances are still getting added without the messages after login.